message_ix_models/tests/tools/costs/test_decay.py

```diff
@@ -21,20 +25,65 @@
         ("cooling", {"coal_ppl__cl_fresh", "gas_cc__air", "nuc_lc__ot_fresh"}),
     ),
 )
-def test_get_cost_reduction_data(module: str, t_exp) -> None:
-    # The function runs without error
-    result = get_cost_reduction_data(module)
+def test_get_module_scenarios_reduction(
+    module: Literal["energy", "materials", "cooling"], t_exp
```

Comment:
Amazing, even with type hints in the tests already :)
If you want to completely type this function: `t_exp` seems to be a `set[str, str, str]` or `set[str, ...]` if you want to be more generic :)

Reply:
thanks @glatterf42 ! :)

i tried to add type hints for `t_exp` -- both `set[str, str, str]` and `set[str, ...]` bring up `mypy` linting errors, but `set[str]` seems to work, so hopefully that's okay?

Reply:
Of course, I was confusing `set` and `tuple` :man_facepalming: